src/models/DefaultPlugin.js

```diff
@@ -190,6 +192,69 @@ class DefaultPlugin {
   repositionComponent(componentId) {
     this.__drawer.repositionComponent(componentId);
   }
+
+  /**
+   * Get the coordinates to use for component draw options.
+   * @param {string} rootId - Html id, without '#'.
+   * @param {ComponentDrawOption} drawOption - The options used to draw this Component.
+   * @param {number} drawOption.x - Position X.
+   * @param {number} drawOption.y - Position Y.
+   * @returns {object} The coordinates for new component.
+   */
+  getComponentPosition(rootId, { x, y }) {
+    const { scale, translate } = this.__drawer.actions.zoom;
+    const { left, top } = document.querySelector(`#${rootId}`).getBoundingClientRect();
+
+    return {
+      x: ((x - left) - translate.x) / scale,
+      y: ((y - top) - translate.y) / scale,
+    };
+  }
+
+  /**
+   * Create new component.
+   * @param {string} rootId - Html id, without '#'.
+   * @param {ComponentDefinition} definition - Component definition.
+   * @param {string} [folder] - Folder path.
+   * @param {string} [fileName] - File name.
```

Comment:
remove `[`...`]` for folder and fileName

Reply:
done